src/records.zig

```diff
@@ -158,13 +163,13 @@ const Line = extern struct {
         return @ptrCast(line.ptr);
     }
 
-    fn convertToAtomRecord(self: *const Line, serialIndex: u32) !AtomRecord {
+    fn convertToAtomRecord(self: *const Line, serialIndex: u32, allocator: std.mem.Allocator) !AtomRecord {
         var atom: AtomRecord = undefined;
         atom.record = strings.removeSpaces(&self.record);
```

Comment:
Change to `try allocator.dupe(u8, strings.removeSpaces(&self.record)` for now. We will remove this later when this is refactored to use composition to infer record type.

Reply:
Ah i see i missed one.  That explains the now commented out segfault in free() on that field.  Not sure if you want to possibly merge any of this, but I'll push these changes now.